platform/android/java/app/src/com/godot/game/GodotApp.java

```diff
@@ -56,8 +64,36 @@ public class GodotApp extends GodotActivity {
 		}
 	}
 
+	// Shouldn't conflict with BaseGodotEditor.RUN_GAME_INFO etc
+	private final static int WINDOW_ID = 668;
+	private final static String TAG = GodotApp.class.getSimpleName();
+
+	private ArrayList<String> commandLineParams = new ArrayList<String>();
+
+	@Override
+	public int onNewGodotInstanceRequested(String[] args) {
+		// Launch a new activity
+		Log.d(TAG, "Restarting with parameters " + String.join(",", args));
+		Godot godot = getGodot();
+		Intent newInstance = new Intent()
+									 .setComponent(new ComponentName(this, GodotApp.class))
+									 .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+									 .putExtra(getEXTRA_COMMAND_LINE_PARAMS(), args);
```

Comment:
You should be able to use `EXTRA_LINE_COMMAND_PARAMS` like a static variable.

Reply:
Nope, this is how interop works :(
I don't daily drive kotlin or java so it took a bit to figure out why it was giving me the "variable has private access" error
I have a better idea for this class anyways.

I figure it's a breaking change to delete `org.godotengine.godot.GodotApp`, but what if it was just an empty subclass of `org.godot.game.GodotApp` ? which is implemented in Kotlin (with a deprecation notice for godot 5)